.github/workflows/macos.yml

```diff
@@ -13,6 +13,17 @@ on:
     # https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks#handling-skipped-but-required-checks
   merge_group:
 
+env:
+  # GITHUB_PULL_REQUEST_URL are used for commenting test reports in Launchable Github App.
+  # https://github.com/launchableinc/cli/blob/v1.80.1/launchable/utils/link.py#L42
+  GITHUB_PULL_REQUEST_URL: ${{ github.event.pull_request.html_url }}
+  # The following envs are necessary in Launchable tokenless authentication.
+  # https://github.com/launchableinc/cli/blob/v1.80.1/launchable/utils/authentication.py#L20
+  LAUNCHABLE_ORGANIZATION: "ruby"
+  LAUNCHABLE_WORKSPACE: "ruby"
+  # https://github.com/launchableinc/cli/blob/v1.80.1/launchable/utils/authentication.py#L71
+  GITHUB_PR_HEAD_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
```

Comment:
Just setting `GITHUB_PR_HEAD_SHA: ${{ github.sha }}` without `github.event.pull_request.head.sha` doesn't work on the pull-requests?
I checked [this page](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) for the `github.sha`.

I see there are [some cases](https://github.com/ruby/ruby/blob/fc2c128e7e1fd4a197669d2f04e8457f4188b7f3/.github/workflows/windows.yml#L101) that the `github.sha` is always used even in the pull-requests in the `.github/workflows/windows.yml`.

Reply:
It doesn't function properly in tokenless authentication. `github.sha` refers to the last virtual merge commit on the branch in [the pull_request event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) and the actual merged commit in [the push event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push). Launchable requires an actual commit that is checked in the Github workflow for tokenless authentication. Thus, we need to give the value of `github.event.pull_request.head.sha` in the pull request, but we need `github.sha` in the main branch.

Reply:
All right. Thanks for explaining the context.